prymer/thermo.py

```diff
@@ -0,0 +1,157 @@
+        """Transfers the values to the underlying implementation class."""
+        self._thermo.set_thermo_args(
+            mv_conc=self.mv_conc_mm,
+            dv_conc=self.dv_conc_mm,
+            dntp_conc=self.dntp_conc_mm,
+            dna_conc=self.dna_conc_nm,
+            dmso_conc=self.dmso_conc_pct,
+            dmso_fact=self.dmso_correction_factor,
+            formamide_conc=self.formamide_conc_mol,
+            annealing_temp_c=self.annealing_temp_c,
+            temp_c=self.temp_c,
+            max_nn_length=self.max_nn_length,
+            max_loop=self.max_loop,
+            tm_method=self.tm_method,
+            salt_corrections_method=self.salt_corrections_method,
+        )
+
+    def tm(self, bases: str) -> float:
+        """
+        Calculates the melting temperature of a sequence of bases.  The calculation used differs
+        for short sequences (len < `max_nn_length`) and long sequences.
+
+        Arguments:
+            bases: a sequence of unambiguous DNA bases (i.e. A/C/G/T)
+
+        Returns:
+            the melting temperature of the sequence, in degrees Celsius
+        """
+        return float(self._thermo.calc_tm(bases))
+
+    def hairpin_tm(self, bases: str) -> float:
+        """
+        Calculates the melting temperature of the most likely secondary structure or hairpin of a
+        single oligo.
+
+        Arguments:
+            bases: a sequence of unambiguous DNA bases (i.e. A/C/G/T)
+
+        Returns:
+            the melting temperature of the most likely (lowest energy) secondary structure
+        """
+        result: ThermoResult = self._thermo.calc_hairpin(bases)
+        return float(result.tm)
+
+    def homodimer_tm(self, bases: str) -> float:
+        """
+        Calculates the melting temperature of the most likely annealing of a single oligo to a
+        second copy of the same oligo (i.e. a duplex of two of the same oligo).
+
+        Arguments:
+            bases: a sequence of unambiguous DNA bases (i.e. A/C/G/T)
+
+        Returns:
+            the melting temperature of the most likely (lowest energy) duplex
+        """
+        result: ThermoResult = self._thermo.calc_homodimer(bases)
+        return float(result.tm)
+
+    def heterodimer_tm(self, bases1: str, bases2: str) -> float:
+        """
+        Calculates the melting temperature of the most likely annealing of two oligos.  The two
+        oligos do *not* need to be the same length.
+
+        Arguments:
+            bases1: a sequence of unambiguous DNA bases (i.e. A/C/G/T) for the first oligo
+            bases2: a sequence of unambiguous DNA bases (i.e. A/C/G/T) for the second oligo
+
+        Returns:
+            the melting temperature of the most likely (lowest energy) heterodimer
+        """
+
+        result: ThermoResult = self._thermo.calc_heterodimer(bases1, bases2)
+        return float(result.tm)
+
+    def heterodimer_3p_anchored_tm(self, bases1: str, bases2: str) -> float:
```

Comment:
why not a boolean called `3p_anchored` in `heterodimer_tm`?

Reply:
I think it's a coin flip.  You could also ask "why not a single `dimer_tm()` that takes an _optional_ second sequence.  At the end of the day I think it's clearer to have separate methods with separate documentation that we can elaborate on.